snapcraft/main.py

```diff
@@ -289,11 +292,13 @@ def _run_clean(args, project_options):
 def _is_store_command(args):
     commands = (
         'list-keys', 'keys', 'register-key',
-        'register', 'upload', 'release', 'push')
+        'register', 'upload', 'release', 'push', 'validate', 'gated')
     return any(args.get(command) for command in commands)
 
 
-def _run_store_command(args):
+# This function's complexity is correlated to the number of
+# commands, no point in checking that.
```

Comment:
well, actually ;)
I think it could be nice to split the commands that handle keys from the commands that handle snap uploads and releases. Not a big deal though, just if you agree.